### PR TITLE
fix being able to add overlapping leaves

### LINKED ISSUE #83: [BUG] Fix being able to add overlapping leaves

### PR DESCRIPTION
**Pull request by Ben-Hanan**
Fixed a bug in being able to add leaves with overlapping periods.
e.g addleave l/d:08/10/2020 d:10/10/2020 l/d:09/10/2020
The above command would not have resulted in an error and added the two leaves as different records of leave but it errors now.

Also made the UI automatically update with any change to leave.

### CODE REVIEW

File: src/main/java/com/eva/model/Model.java [hunk 1]
```diff
@@ -115,11 +115,17 @@
...
     /**
-     * Returns true if a staff with the same identity as {@code staff} has the same leave as {@code leave}.
-     * @return
+     * Returns the conflicting leave if a staff with the same identity as {@code staff}
+     * has the same leave period as {@code leave}.
      */
     Optional<Leave> hasLeaveDate(Staff target, LocalDate date);
 
+    /**
+     * Returns the conflicting leave if a staff with the same identity as {@code staff}
```
Comment (Nikhilalalalala) [inline src/main/java/com/eva/model/Model.java]: Minor, but the javadoc is a little misleads readers to think the specific leave object that is conflicting will be returned.

File: src/main/java/com/eva/commons/util/DateUtil.java [hunk 2]
```diff
@@ -74,8 +75,19 @@ public static String dateToString(LocalDate date) {
...
      */
     public static int getDaysBetween(LocalDate start, LocalDate end) {
-        requireNonNull(start);
-        requireNonNull(end);
+        requireAllNonNull(start, end);
         return Period.between(start, end).getDays() + 1;
     }
+
+    /**
+     * Returns a list of LocalDates between the two given {@code dates}.
```
Comment (Nikhilalalalala) [inline src/main/java/com/eva/commons/util/DateUtil.java:83]: Minor, Good to specify if the list returned is inclusive or exclusive of the start and end date

**Comment by Nikhilalalalala** [comment]
LGTM! Just some minor comments on the javadocs part. Nice observation on this bug :)

**Comment by TheSpaceCuber** [comment]
LGTM

**Pull request merged**